### PR TITLE
Switch to config.json

### PR DESCRIPTION
**Pull request by kimpepper**

### CODE REVIEW

**Comment by nicksantamaria** [comment]
Please update `README.md` to show how it should be used.

File: config.go [hunk 1]
```diff
@@ -1,121 +1,101 @@
...
-		return fallback
+	if _, err := os.Stat(config.path); os.IsNotExist(err) {
+		return config, errors.Wrap(err, "config file does not exist")
 	}
 
-	return value
-}
+	data, err := ioutil.ReadFile(config.path)
+	if err != nil {
+		return config, errors.Wrap(err, "Failed to read config file")
```
Comment (nickschuch) [inline config.go]: lower case the F

File: config.go [hunk 2]
```diff
@@ -1,121 +1,101 @@
...
-	//
-	// @see https://github.com/skpr/docs/blob/master/docs/config.md#deep-dive
-	paths := []string{
-		c.filePath(PathPartConfig, PathPartDefault, key),
-		c.filePath(PathPartSecret, PathPartDefault, key),
-		c.filePath(PathPartConfig, PathPartOverride, key),
-		c.filePath(PathPartSecret, PathPartOverride, key),
+	err = json.Unmarshal(data, &configData)
+	if err != nil {
+		return config, errors.Wrap(err, "Failed to unmarshal config")
```
Comment (nickschuch) [inline config.go]: lower case the F

File: config.go [hunk 3]
```diff
@@ -1,121 +1,101 @@
...
+}
 
-	return strings.TrimSuffix(value, DefaultTrimSuffix), nil
+// GetInt returns a int value for the key.
+func (c *Config) GetInt(key string) (int, bool) {
+	value, ok := c.getValue(key)
+	if value == nil {
+		value = 0
+	}
+	return int(value.(float64)), ok
```
Comment (nickschuch) [inline config.go]: should this be an `int`?

File: config.go [hunk 4]
```diff
@@ -1,121 +1,101 @@
...
+	value, ok := c.getValue(key)
+	if value == nil {
+		value = false
 	}
+	return value.(bool), ok
+}
 
-	return strings.TrimSuffix(value, DefaultTrimSuffix), nil
+// GetInt returns a int value for the key.
+func (c *Config) GetInt(key string) (int, bool) {
```
Comment (nickschuch) [inline config.go]: Maybe we could make these return an `error` instead. That way we could:
* Use https://golang.org/pkg/strconv/#Atoi
* Have an error of type "Not found" which we could check for.

Comment (kimpepper) [reply config.go]: Discussed with @nickschuch and agreed it's ok to return the ok as a boolean.

**Pull request merged**